src/gui/torrentcategorydialog.cpp

```diff
@@ -128,12 +135,62 @@ void TorrentCategoryDialog::setCategoryName(const QString &categoryName)
     m_ui->textCategoryName->setText(categoryName);
 }
 
-QString TorrentCategoryDialog::savePath() const
+BitTorrent::CategoryOptions TorrentCategoryDialog::categoryOptions() const
+{
+    BitTorrent::CategoryOptions categoryOptions;
+    categoryOptions.savePath = m_ui->comboSavePath->selectedPath();
+    if (m_ui->comboUseDownloadPath->currentIndex() == 1)
```

Comment:
use `switch ()`?

Reply:
> use `switch ()`?

I would prefer not to use `switch` when there are only a couple of cases here, especially when we don't have to handle them all.